File: SoftUni - Advanced/Defining Classes/Defining Classes - Exercise/08. Car Salesman/Car.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace _8.Car_Salesman
{
    public class Car
    {
        public string  Model { get; set; }

        public Engine Engine { get; set; }

        public int Weight { get; set; }

        public string  Color { get; set; }

        public Car(string model, Engine engine)
        {
            Model = model;
            Engine = engine;
        }

        public override string ToString()
        {
            var sb = new StringBuilder();
            sb.Append($"{Model}:");
            sb.AppendLine();
            sb.Append(Engine.ToString());
            sb.AppendLine();
            sb.Append(Weight == 0 ? "  Weight: n/a" : $"  Weight: {Weight}");
            sb.AppendLine();
            sb.Append(string.IsNullOrEmpty(Color) ? "  Color: n/a" : $"  Color: {Color}");
            return sb.ToString().TrimEnd();
        }
    }
}

File: SoftUni - Java Advanced/Defining Classes/Define Classes - Exercise/_05_Car_Salesman/Car.java
package exercise._05_Car_Salesman;

public class Car {
    private String model;
    private Engine engine; // Reference to an Engine object
    private String weight;
    private String color;

    public Car(String model, Engine engine, String weight, String color) {
        this.model = model;
        this.engine = engine;
        this.weight = weight;
        this.color = color;
    }

    public Car(String model, Engine engine) {
        this(model, engine, "n/a", "n/a");
    }

    public Car(String model, Engine engine, String weight) {
        this(model, engine, weight, "n/a");
    }

    @Override
    public String toString() {
        return model + ":\n" +
                engine + "\n" + // Calls the Engine's toString method
                "Weight: " + weight + "\n" +
                "Color: " + color;
    }
}
